src/game.rs

```diff
@@ -41,6 +41,12 @@ impl Game {
     }
 
     pub fn move_piece(&mut self, from: Position, to: Position) -> ChessResult<()> {
-        self.board.move_piece(from, to)
+        self.board.move_piece(from, to)?;
+        self.board.player = match self.board.player {
+            Player::White => Player::Black,
+            Player::Black => Player::White,
+        };
+        self.board.add_moves();
```

Comment:
Should we have `move_piece()` in `Board` call `add_moves()` immediately after moving the piece? So that the game isn't responsible for updating the moves, and anytime the board state changes it's guaranteed to have the moves updated

Reply:
Yeah I was trying to make this kinda model, view, controller, but I totally get what you're saying. If make it `board`'s responsibility, `game` is pretty much entirely a wrapper.  There are potential reasons to want to move without re-adding the moves and changing the player, like for testing if a move puts you in check. Not sure what the best option is.

Reply:
Ok yeah so if we want Game is manage the turns then we can make the `next_turn()` (or `end_turn()`) in game, doing the same logic: Once we move in game, then we calculate new moves and swap the players for current turn.

Reply:
We could just write a function that checks if a move puts you in check right?